common/achievement/get.ts

```diff
@@ -0,0 +1,99 @@
+    };
+    const newContextValue = await Promise.all(
+        contextKeys.map(async (key) => {
+            const [type, id] = userAchievements[currentAchievementIndex].context[key].split('/');
+            const newContextValue = await prisma[type].findUnique({
+                where: { id: Number(id) },
+            });
+            return { key: type, value: newContextValue };
+        })
+    );
+    newContextValue.forEach((context) => {
+        achievementContext[context.key] = context.value;
+    });
+    const currentAchievementTemplate = userAchievements[currentAchievementIndex].template as Achievement_template;
+    const templateKeys = Object.keys(userAchievements[currentAchievementIndex].template);
+    templateKeys.forEach((key) => {
+        const updatedElement =
+            currentAchievementTemplate[key] && typeof currentAchievementTemplate[key] === 'string'
+                ? renderTemplate(currentAchievementTemplate[key], achievementContext)
+                : currentAchievementTemplate[key];
+        currentAchievementTemplate[key] = updatedElement;
+    });
+    const resultIndex = currentAchievementIndex < 0 ? null : currentAchievementIndex;
+    const state: achievement_state =
+        userAchievements.length === 0
+            ? achievement_state.INACTIVE
+            : userAchievements[currentAchievementIndex].achievedAt
+            ? achievement_state.COMPLETED
+            : achievement_state.ACTIVE;
+    const newAchievement = state === achievement_state.COMPLETED && !userAchievements[resultIndex].isSeen;
+
+    return {
+        name: currentAchievementTemplate.name,
+        subtitle: currentAchievementTemplate.subtitle,
+        description: currentAchievementTemplate.description,
+        image: currentAchievementTemplate.image,
+        alternativeText: 'alternativeText',
+        actionType: currentAchievementTemplate.actionType as achievement_action_type_enum,
+        achievementType: currentAchievementTemplate.type as achievement_type_enum,
+        achievementState: state,
+        steps: currentAchievementTemplate.stepName
+            ? userAchievements.map((achievement, index) => {
+                  // if a achievementTemplate has a stepName, it means that it must have multiple steps as well as being a sequential achievement
+                  // for every achievement in the sortedGroupAchievements, we create a step object with the stepName (descirption) and isActive property for the achievement step currently active but unachieved
+                  return {
+                      description: achievement.template.stepName,
+                      isActive: index === resultIndex,
+                  };
+              })
+            : null,
+        maxSteps: userAchievements.length,
+        currentStep: userAchievements.length,
+        newAchievement: newAchievement,
+        progressDescription: `Noch ${userAchievements.length - userAchievements.length} Schritte bis zum Abschluss`,
```

Comment:
This was my first thought too but the problem here is, that we do not know the progressDescription in the frontend. It could be "Noch X Schritte" or "Noch X Meetings" or anything else. The unit is never certain. I completely forgot about this issue but I think we need an extra field for the progress bar description that could handle this. Maybe I am wrong, let me know your thoughts on this!